article/app/services/MessageUsService.scala

```diff
@@ -0,0 +1,50 @@
+package services
+
+import common.{Box, GuLogging}
+import model.MessageUsConfigData
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class MessageUsService(messageUsS3Client: S3Client[MessageUsConfigData]) extends GuLogging {
+
+  private val MAX_LIVEBLOGS_WITH_MESSAGE_US = 50
+  private val messageUsConfigData = Box[Option[Map[String, MessageUsConfigData]]](None)
+
+  def refreshMessageUsData()(implicit executionContext: ExecutionContext): Future[Unit] = {
+    val listOfKeys = messageUsS3Client
+      .getListOfKeys()
+      .map(keys => {
+        if (keys.length > MAX_LIVEBLOGS_WITH_MESSAGE_US)
+          log.warn(s"Over 50 live blogs are stored in S3, only caching the first 50 and ignoring the rest!")
```

Comment:
Is there a reason for this limit? how is it managed on the backend?

Reply:
In theory this would never get to 50 items because we are managing the S3 data in a way that it would only pick the running liveblogs which were created in the last 2 days and have the `Message Us` feature enabled. The code for this can be found here https://github.com/guardian/editorial-experience-lambda/blob/main/message-us-worker/src/main/scala/com/gu/messageusworker/clients/ContentApiClient.scala#L25

But since frontend is a critical service and to avoid putting any risk of overwhelming memory by too much data, we thought it'd be good to have a limit for the number of items to store in the cache. The frontend cache has far more capacity than what is needed for 50 items, but because in practice, it should never happen that there are more than 50 running blogs with `Message Us` enabled, we thought it'd be a good max number. 

But I'm more than happy to hear if you have any more thoughts on this :)

Reply:
Unfortunately I do not have access to that particular github project, but I will take your word for it! All good - was just helpful to understand that the limitation is applied somewhere that wouldn't mean people are unsure of the intended functionality in the future! Thanks :)